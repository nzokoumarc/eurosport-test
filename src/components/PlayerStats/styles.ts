import { makeStyles, Theme } from '@material-ui/core';

export const useStyles = makeStyles((theme: Theme) => ({
  flex: {
    display: 'flex',
    borderTop: '1px solid #aaa',
    padding: theme.spacing(1, 0),
    justifyContent: 'space-around',
    margin: theme.spacing(1, 0),
  },
}));
