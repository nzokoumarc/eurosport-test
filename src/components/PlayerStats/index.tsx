import * as React from 'react';
import { Typography } from '@material-ui/core';
import { DataType, getHeight, getWeight, getScore } from '../../utils/players';

import { useStyles } from './styles';

type PlayerStatsProps = {
  data: DataType;
};

const PlayerStats: React.FC<PlayerStatsProps> = React.memo(({ data }) => {
  const classes = useStyles('');
  return (
    <>
      <div className={classes.flex}>
        <div>
          <Typography color="initial" variant="body2">
            Height (M)
          </Typography>
          <Typography color="primary" variant="h6">
            {getHeight(data.height)}
          </Typography>
        </div>
        <div>
          <Typography color="initial" variant="body2">
            Weight (kg)
          </Typography>
          <Typography color="primary" variant="h6">
            {getWeight(data.weight)}
          </Typography>
        </div>
        <div>
          <Typography color="initial" variant="body2">
            Age
          </Typography>
          <Typography color="primary" variant="h6">
            {data.age}
          </Typography>
        </div>
        <div>
          <Typography color="initial" variant="body2">
            Rank
          </Typography>
          <Typography color="primary" variant="h6">
            {data.rank}
          </Typography>
        </div>
        <div>
          <Typography color="initial" variant="body2">
            Win
          </Typography>
          <Typography color="primary" variant="h6">
            {getScore(1, data.last)}
          </Typography>
        </div>
        <div>
          <Typography color="initial" variant="body2">
            Lost
          </Typography>
          <Typography color="primary" variant="h6">
            {getScore(0, data.last)}
          </Typography>
        </div>
      </div>
    </>
  );
});

export default PlayerStats;
