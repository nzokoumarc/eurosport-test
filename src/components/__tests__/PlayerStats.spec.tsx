import * as React from 'react';
import { shallow } from 'enzyme';

import Mocks from '../__mocks__';
import PlayerStats from '../PlayerStats';

describe('PlayerStats', () => {
  it('renders', () => {
    const component = shallow(<PlayerStats data={Mocks.data} />);
    expect(component).toMatchSnapshot();
  });
});
