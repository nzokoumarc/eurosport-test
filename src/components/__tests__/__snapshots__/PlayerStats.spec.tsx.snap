// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerStats renders 1`] = `
<Fragment>
  <div
    className="makeStyles-flex-1"
  >
    <div>
      <WithStyles(ForwardRef(Typography))
        color="initial"
        variant="body2"
      >
        Height (M)
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        color="primary"
        variant="h6"
      >
        1.83
      </WithStyles(ForwardRef(Typography))>
    </div>
    <div>
      <WithStyles(ForwardRef(Typography))
        color="initial"
        variant="body2"
      >
        Weight (kg)
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        color="primary"
        variant="h6"
      >
        81
      </WithStyles(ForwardRef(Typography))>
    </div>
    <div>
      <WithStyles(ForwardRef(Typography))
        color="initial"
        variant="body2"
      >
        Age
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        color="primary"
        variant="h6"
      >
        33
      </WithStyles(ForwardRef(Typography))>
    </div>
    <div>
      <WithStyles(ForwardRef(Typography))
        color="initial"
        variant="body2"
      >
        Rank
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        color="primary"
        variant="h6"
      >
        21
      </WithStyles(ForwardRef(Typography))>
    </div>
    <div>
      <WithStyles(ForwardRef(Typography))
        color="initial"
        variant="body2"
      >
        Win
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        color="primary"
        variant="h6"
      >
        4
      </WithStyles(ForwardRef(Typography))>
    </div>
    <div>
      <WithStyles(ForwardRef(Typography))
        color="initial"
        variant="body2"
      >
        Lost
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        color="primary"
        variant="h6"
      >
        1
      </WithStyles(ForwardRef(Typography))>
    </div>
  </div>
</Fragment>
`;
